### PR TITLE
add custom annotations in the tidb-operator helm chart (#3957)

### LINKED ISSUE #3957: Support for setting custom annotations on tidb-scheduler pod in tidb-operator helm chart
## Feature Request

**Is your feature request related to a problem? Please describe:**

The Datadog agent detects the `tidb-scheduler` pod as the standard k8s scheduler, since it uses the upstream image. However, it uses a custom port, so the automatic Datadog agent config for it doesn't work. I need to be able to specify annotations on the pod to override the automatic Datadog agent config.

**Describe the feature you'd like:**

It should be possible to set `scheduler.podAnnotations` in the helm chart to apply custom annotations to the `tidb-scheduler` pod.

**Describe alternatives you've considered:**

The only currently alternatives are forking the helm chart or a post-renderer script, neither of which is a good solution in my environment.

**Teachability, Documentation, Adoption, Migration Strategy:**

The need to set custom annotations on pods is a relatively common one.

### PR DESCRIPTION
**Pull request by max-f-z**
### What problem does this PR solve?

feat: helm chart add support for setting custom pod annotations for scheduler
Closes #3957 

### What is changed and how does it work?

in helm chart provide scheduler.podAnnotations, and go along with helm install

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [x] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: charts/tidb-operator/values.yaml [hunk 1]
```diff
@@ -153,6 +153,9 @@ scheduler:
   # runAsUser: 1000
   # runAsGroup: 2000
   # fsGroup: 2000
+  podAnnotations: {}
+  # prometheus.io/scrape: "true"
+  # prometheus.io/port: "9090"
```
Comment (DanielZhangQD) [inline charts/tidb-operator/values.yaml:158]:
```suggestion
  # Refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
  annotations: {}
```

Comment (DanielZhangQD) [reply charts/tidb-operator/values.yaml:158]: Suggest also adding the annotations for the `controller-manager` section.

File: charts/tidb-operator/templates/scheduler-deployment.yaml [hunk 2]
```diff
@@ -26,6 +26,10 @@ spec:
         app.kubernetes.io/name: {{ template "chart.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
         app.kubernetes.io/component: scheduler
+      annotations:
```
Comment (csuzhangxc) [inline charts/tidb-operator/templates/scheduler-deployment.yaml:29]: How about putting this line into the `if` block?

Comment (max-f-z) [reply charts/tidb-operator/templates/scheduler-deployment.yaml:29]: Seems better, will do that in a bit

File: charts/tidb-operator/values.yaml (hunk 1 above)
Comment (DanielZhangQD) [reply charts/tidb-operator/values.yaml:158]: @max-f-z I just revert the parameter name, the original name is good.

Comment (max-f-z) [reply charts/tidb-operator/values.yaml:158]: ok, will add annotations for controller-manager as well, thanks @DanielZhangQD

File: charts/tidb-operator/values.yaml [hunk 3]
```diff
@@ -112,6 +112,8 @@ controllerManager:
   # runAsUser: 1000
   # runAsGroup: 2000
   # fsGroup: 2000
+  podAnnotations: {}
+  # Refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
```
Comment (csuzhangxc) [inline charts/tidb-operator/values.yaml:116]: We often put the refer above the fields, ie

```yaml
  # Refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
  podAnnotations: {}
```

Comment (max-f-z) [reply charts/tidb-operator/values.yaml]: sure

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by max-f-z** [comment]
seems like some ci checks are not triggered. Or am i missing something here to get this merged?
Please help, @csuzhangxc

**Pull request merged**